src/ServiceControl/Upgrade/StaleIndexAfterUpgrade.cs

```diff
@@ -0,0 +1,84 @@
+﻿namespace Particular.ServiceControl.Upgrade
+{
+    using System;
+    using System.IO;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using NServiceBus;
+    using NServiceBus.Features;
+
+    public class StaleIndexAfterUpgrade : Feature
+    {
+        public StaleIndexAfterUpgrade()
+        {
+            EnableByDefault();
```

Comment:
How does the feature pipeline work, could we check for the existence of the file at this point already?

Reply:
We could do it in the conditions but I'm not sure whether it is really better in terms of having a feature disabled vs a feature enabled but the startup task noop. Happy to change though

Reply:
The SC log lists all the features at startup. Excluding it here would be different to a noop in that the feature would no longer appear in the log file at all.

I'm trying to decide whether or not that's a good thing...

Reply:
I think it would still appear but it would be listed as disabled. I think it's fine as is.